### PR TITLE
feat: initial migration

### PR DESCRIPTION
**Pull request by UncleGedd**
initial migration of runner code from UDS CLI

### CODE REVIEW

File: src/cmd/run.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	zarfUtils "github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/exec"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/helpers"
+	"github.com/pterm/pterm"
+	"github.com/spf13/cobra"
+)
+
+var runCmd = &cobra.Command{
+	Use: "run",
+	PersistentPreRun: func(cmd *cobra.Command, _ []string) {
```
Comment (decleaver) [inline src/cmd/run.go:26]:
```suggestion
	PersistentPreRun: func(_ *cobra.Command, _ []string) {
```

File: src/cmd/run.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		}
+		return taskNames, cobra.ShellCompDirectiveNoFileComp
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		if len(args) > 1 && !config.ListTasks {
+			return fmt.Errorf("accepts 0 or 1 arg(s), received %d", len(args))
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
```
Comment (decleaver) [inline src/cmd/run.go:55]:
```suggestion
	Run: func(_ *cobra.Command, args []string) {
```

**Review by decleaver — COMMENTED** [review]
looks good! just a couple gh linting suggestions

**Review by decleaver — APPROVED** [review]
Maru!!!

**Review by mjnagel — APPROVED** [review]
⛵

**Pull request merged**